File: toplas2021/benchmarks/seahorn/03-prusti/prusti-3-pass-paper_examples-points-compress.c
#include "seahorn/seahorn.h"
#include <stdlib.h>

typedef struct Point {
  int x, y;
} Point;
typedef struct Pair {
  Point *_0, *_1;
} Pair;

extern Point nd();

void shift_x(Point *p, int s) { p->x = p->x + s; }

int main() {
  Pair segm = {(Point *)malloc(sizeof(Point)), (Point *)malloc(sizeof(Point))};
  *segm._0 = nd();
  *segm._1 = nd();
  int diff = segm._0->x - segm._1->x + 1;
  shift_x(&segm._1, diff);
  sassert(segm._0->x < segm._1->x);
  return 0;
}

File: toplas2021/benchmarks/rust-horn/03-prusti/prusti-3-pass-paper_examples-points-compress.rs
fn rand<T>() -> T { unimplemented!() }

struct Point {
  x: i32,
  y: i32,
}

fn shift_x(p: &mut Point, s: i32) { p.x = p.x + s }

fn main() {
  let mut segm: (Box<Point>, Box<Point>) = rand();
  let diff = (*segm.0).x - (*segm.1).x + 1;
  shift_x(&mut segm.1, diff);
  assert!((*segm.0).x < (*segm.1).x);
}
